src/ring_theory/integrally_closed.lean

```diff
@@ -75,18 +75,25 @@ end iff
 
 namespace is_integrally_closed
 
-variables {R : Type*} [comm_ring R] [is_domain R] [iic : is_integrally_closed R]
-variables {K : Type*} [field K] [algebra R K] [is_fraction_ring R K]
+variables {R : Type*} [comm_ring R] [id : is_domain R] [iic : is_integrally_closed R]
+variables {K : Type*} [field K] [algebra R K] [ifr : is_fraction_ring R K]
 
-instance : is_integral_closure R R K :=
-(is_integrally_closed_iff_is_integral_closure K).mp iic
+include iic ifr
 
-include iic
+instance : is_integral_closure R R K := (is_integrally_closed_iff_is_integral_closure K).mp iic
 
 lemma is_integral_iff {x : K} : is_integral R x ↔ ∃ y, algebra_map R K y = x :=
 is_integral_closure.is_integral_iff
 
-omit iic
+omit iic ifr
+
+lemma exists_algebra_map_eq_of_pow_mem {S : subalgebra R K} [is_domain S] [is_integrally_closed S]
```

Comment:
I can prove the `is_scalar_tower R S K` version (which like you said, the `R` doesn't play any role here):
```
lemma exists_algebra_map_eq_of_is_integral_pow {x : K} {n : ℕ} (hn : 0 < n)
  (hx : is_integral S $ x ^ n) : ∃ y : S, algebra_map S K y = x :=
is_integral_iff.mp $ is_integral_of_pow hn hx
```
This is an immediate application of `is_integral_iff` and `is_integral_of_pow` that I proved elsewhere, so I didn't think it was necessary to have this, but I am happy to include it.

But I don't seem to be able to prove the `subalgebra` version from this, because I believe a `subalgebra` is not a `comm_ring`. This is the form I need, since I have `x ∈ S`, which cannot be expressed if `S` is only a `Type*` with a `comm_ring S` instance. My use case happens to be both a `subalgebra` and a `comm_ring`, but it's inconvenient to use the assumption `is_integral S $ x ^ n` as above rather than `x ∈ S` that I already have.

Reply:
Hmm I'm not sure what I did before, but I am now able to prove the subalgebra version from the above, right after I sent the previous message - I'll push my changes.